.github/workflows/bazel.yml

```diff
@@ -353,49 +495,38 @@ jobs:
+            printf '\r%s\r' '###############################' 1>&2
+            printf '\r%s\r' 'Configuring bazel remote cache.' 1>&2
+            printf '\r%s\r\r' '###############################' 1>&2
+            bazel/scripts/remote_cache_bazelrc_setup.sh "${{ env.CACHE_KEY }}" "${{ env.REMOTE_DOWNLOAD_OPTIMIZATION }}" "${{ secrets.BAZEL_REMOTE_PASSWORD }}"
 
             printf '\r%s\r' '###############################' 1>&2
             printf '\r%s\r' 'Determining bazel test targets.' 1>&2
             printf '\r%s\r' '###############################' 1>&2
-            # Check that the workflow is run on a PR and that the .bazelrc and WORKSPACE files have not been affected.
-            if [[ ("${{ github.event_name }}" == "pull_request") && \
-                  ! $(git diff-tree --no-commit-id --name-only -r "${{ github.event.pull_request.base.sha }}" "${{ github.SHA }}" | \
-                    grep -E '(.bazelrc|WORKSPACE.bazel)') \
-                ]];
+            if [[ "${{ needs.bazel_diff.outputs.bazel_diff_ran }}" == 'true' ]];
             then
-              printf '\r%s\r' 'Determining targets via bazel-diff ...' 1>&2
-              IMPACTED_TARGETS_FILE="/tmp/impacted_targets.txt"
-              bazel/scripts/bazel_diff.sh "${{ github.event.pull_request.base.sha }}"  "${{ github.SHA }}" | tee "$IMPACTED_TARGETS_FILE"
-              printf '\r%s\r' 'Determining impacted targets ...' 1>&2
-              # Choose all impacted targets of type 'py_binary' that are tagged as service.
-              TEST_TARGETS=$(bazel/scripts/filter_test_targets.sh "py_binary" "service" < "$IMPACTED_TARGETS_FILE")
+              TEST_TARGETS="${{ needs.bazel_diff.outputs.py_service_targets }}"
+              printf '\r%s\r' 'The service test targets of type py_binary are:' 1>&2
               printf '\r%s\r' "$TEST_TARGETS" 1>&2
             else
               printf '\r%s\r' 'Checking all service targets of type "py_binary" for ModuleNotFoundError.' 1>&2
               TEST_TARGETS="ALL_PYTHON_SERVICE_TARGETS"
             fi
 
-            printf '\r%s\r' '###############################' 1>&2
-            printf '\r%s\r' 'Configuring bazel remote cache.' 1>&2
-            printf '\r%s\r\r' '###############################' 1>&2
-            bazel/scripts/remote_cache_bazelrc_setup.sh "${{ env.CACHE_KEY }}" "${{ env.REMOTE_DOWNLOAD_OPTIMIZATION }}" "${{ secrets.BAZEL_REMOTE_PASSWORD }}"
-
             printf '\r%s\r' '###############################' 1>&2
             printf '\r%s\r' 'Executing python import bazelification check.' 1>&2
             printf '\r%s\r' '###############################' 1>&2
             if [[ -n "${TEST_TARGETS}" ]];
             then
               if [[ "${TEST_TARGETS}" == "ALL_PYTHON_SERVICE_TARGETS" ]];
               then
-                bazel/scripts/test_python_service_imports.sh;
+                bazel/scripts/test_python_service_imports.sh 1>&2;
```

Comment:
These changes are unrelated, but this fixes errors in the ordering of the logging.